code/datums/diseases/advance/advance.dm

```diff
@@ -305,9 +334,15 @@
 		var/datum/reagent/D = GLOB.chemical_reagents_list[cures[1]]
 		cure_text = D.name
 
-// Randomly generate a symptom, has a chance to lose or gain a symptom.
-/datum/disease/advance/proc/Evolve(min_level, max_level, ignore_mutable = FALSE)
-	if(!mutable && !ignore_mutable)
+// Randomly generate a symptom if no symptoms are focused. If symptoms are focused, adds all of them at once.
+/datum/disease/advance/proc/Evolve(datum/reagents/container, min_level, max_level, ignore_mutable = FALSE)
+	if(!container || !mutable && !ignore_mutable)
+		return
+	var/list/focused_symptoms = GetFocusedSymptoms(container, min_level, max_level)
+	if(focused_symptoms.len)
+		for(var/datum/symptom/S as anything in focused_symptoms)
```

Comment:
Do not shorten variable names. Be clear about what this is.

Reply:
Uh, sure.